build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 The IREE Authors
```

Comment:
It would be nice if we could avoid having to write the same script twice here. You can invoke CMake to execute a CMake script, but of course getting docker to use files outside of its context is a whole thing... Really just rambling here, I guess.

Reply:
I don't have a strong opinion here, but I will say this was specifically intended to address https://github.com/google/iree/issues/8971 which specifies using CMake to enable seamless multi-platform support and integrate with the build system.

Reply:
What I think Geoffrey is getting at is that we _could_ also make Docker invoke CMake and drop the non-CMake script. Or we could make `fetch_cuda_deps.sh` itself more portable and run it under CMake. Generally, having the logic duplicated across two files used in different contexts is a bit more work to maintain. I don't really anticipate this being something that we need to update frequently though.

Reply:
On my local machine, the download is pretty fast, never more than 10-20 seconds. Its just downloading some tools header files, not an entire repo. 

Bear in mind, I have seen things on the CI machines that require networks request can randomly fail (see the TFLite tests). There is no retry mechanism here. On a local machine, thats not a huge deal since you can build again.

Reply:
Scott is right that that's what I meant. I'd prefer to minimize network dependencies in the CI